frontend/view-edit-lead/lead-home/view-edit-basic-lead-info.component.tsx

```diff
@@ -1,34 +1,122 @@
 import React from "react";
+import easyFetch from "../../util/easy-fetch";
 import { SingleLead } from "../view-lead.component";
 import LeadSection from "./lead-section.component";
+import BasicLeadInformationInputs from "../edit-lead-inputs/basic-lead-information-inputs.component";
 
 export default function ViewEditBasicLeadInfo(
   props: ViewEditBasicLeadInfoProps
 ) {
+  const [editing, setEditing] = React.useState(false);
+  const [apiStatus, dispatchApiStatus] = React.useReducer(updatingReducer, {
+    isUpdating: false,
+    newLeadData: null
+  });
   const { lead } = props;
 
+  React.useEffect(() => {
+    if (apiStatus.isUpdating) {
+      const abortController = new AbortController();
+      easyFetch(`/api/leads/${props.lead.id}`, {
+        method: "PATCH",
+        body: apiStatus.newLeadData,
+        signal: abortController.signal
+      }).then(data => {
+        dispatchApiStatus({ type: "reset" });
+        props.leadUpdated(data.lead);
+        setEditing(false);
+      });
```

Comment:
```suggestion
      }).catch(err => {
        setTimeout(() => {
          throw err;
        })
      })
```

Reply:
Do we need this in the /view-edit-client/client-home/view-edit-basic-info.component.tsx file's easyFetch? The .catch is missing there.